udata/core/search/views.py

```diff
@@ -9,34 +9,29 @@
 from udata.utils import multi_to_dict
 from udata.features.territories import check_for_territories
 
+# Maps template variables names to model types
+MAPPING = {
+    'datasets': Dataset,
+    'reuses': Reuse,
+    'organizations': Organization,
+    'users': User,
+}
+
 
 @front.route('/search/', endpoint='search')
 def render_search():
     params = multi_to_dict(request.args)
     params['facets'] = True
     # We only fetch relevant data for the given filter.
     if 'tag' in params:
-        search_queries = [
-            search.SearchQuery(Dataset, **params),
-            search.SearchQuery(Reuse, **params)
-        ]
-        results_labels = ['datasets', 'reuses']
+        types = ['datasets', 'reuses']
     elif 'badge' in params:
-        search_queries = [
-            search.SearchQuery(Dataset, **params),
-            search.SearchQuery(Organization, **params)
-        ]
-        results_labels = ['datasets', 'organizations']
+        types = ['datasets', 'organizations']
     else:
-        search_queries = [
-            search.SearchQuery(Dataset, **params),
-            search.SearchQuery(Reuse, **params),
-            search.SearchQuery(Organization, **params),
-            search.SearchQuery(User, **params)
-        ]
-        results_labels = ['datasets', 'reuses', 'organizations', 'users']
-    results = search.multisearch(*search_queries)
-    context = dict(zip(results_labels, results))
+        types = ['datasets', 'reuses', 'organizations', 'users']
+    models = [MAPPING[typ] for typ in types]
```

Comment:
To avoid the mapping, why don't we use directly the models in the lists and then zip with the `__class__.__name__`?